### PR TITLE
set Response Size for HttpClient to 1 MB

### PR DESCRIPTION
**Pull request by SomkaPe**

### CODE REVIEW

File: tests/Test.MSAL.NET.Unit/HttpTests/HttpClientFactoryTests.cs [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Test.MSAL.NET.Unit.Mocks;
+
+namespace Test.MSAL.NET.Unit.HttpTests
+{
+    [TestClass]
+    public class HttpClientFactoryTests
+    {
+        [TestMethod]
+        public void GetHttpClient_MaxRespContentBuffSizeSetTo1Mb()
```
Comment (kpanwar) [inline tests/Test.MSAL.NET.Unit/HttpTests/HttpClientFactoryTests.cs]: You should add a test case that actually generates response greater than 1MB and ensure that the http client fails. See adal v3 for reference

Comment (SomkaPe) [inline tests/Test.MSAL.NET.Unit/HttpTests/HttpClientFactoryTests.cs]: added simple test for public and confidential clients

File: src/MSAL.PCL/Internal/Http/HttpClientFactory.cs [hunk 2]
```diff
@@ -35,7 +35,17 @@ internal class HttpClientFactory
         private static HttpClient _client;
         private static readonly object LockObj = new object();
         public static bool ReturnHttpClientForMocks { set; get; }
+        public const long MaxResponseContentBufferSizeInBytes = 1024*1024;
 
+        private static HttpClient CreateHttpClient()
```
Comment (rayluo) [inline src/MSAL.PCL/Internal/Http/HttpClientFactory.cs]: While the implementation in this PR is functionally correct, I would recommend to change it slightly.

    // The newly implemented CreateHttpClient() can be avoided, if we refactor the GetHttpClient() into something like this
    public static HttpClient GetHttpClient()
    {
        var httpClient = new HttpClient(HttpMessageHandlerFactory.GetMessageHandler(ReturnHttpClientForMocks))
             {
                 MaxResponseContentBufferSize = MaxResponseContentBufferSizeInBytes
             };
        if (ReturnHttpClientForMocks) {
            return httpClient;
        }

        if (_client == null)
        {
            lock (LockObj)
            {
                if (_client == null)
                {
                    _client = httpClient;
                }
            }
        }
    }

The reason for the change: At a hindsight, the preexisting GetHttpClient() contains duplicated code already. The implementation in this PR consolidates them into one, which is good, with the help of a new shared helper function. But we could avoid such extra helper function, if we don't have duplication logic in the first place. Also, aesthetically speaking, current implementation in the PR ends up with 2 static method with similar names, `GetHttpClient()` and `CreateHttpClient()`. Future caller/maintainer may not intuitively notice one of them are private.

Last but not the least, the above implementation minimizes the codes would be run within a lock.

Comment (kpanwar) [inline src/MSAL.PCL/Internal/Http/HttpClientFactory.cs]: @rayluo - your code is functionally incorrect. When mocks are not in play, your suggested code will create a new httpclient every time. But @SomkaPe's implementation keeps the actual intent intact.

File: src/MSAL.PCL/Internal/Http/HttpClientFactory.cs [hunk 3]
```diff
@@ -35,7 +35,17 @@ internal class HttpClientFactory
...
+        {
+            var httpClient = new HttpClient(HttpMessageHandlerFactory.GetMessageHandler(ReturnHttpClientForMocks))
+            {
+                MaxResponseContentBufferSize = MaxResponseContentBufferSizeInBytes
+            };
+
+            return httpClient;
+        }
```
Comment (kpanwar) [inline src/MSAL.PCL/Internal/Http/HttpClientFactory.cs]: insert new line after }.
It is a good practice to run resharper code reformat to align curly braces as well.

File: src/MSAL.PCL/Internal/Http/HttpClientFactory.cs (hunk 2 above)
Comment (rayluo) [inline src/MSAL.PCL/Internal/Http/HttpClientFactory.cs]: You are right. My implementation was based on an assumption that creating an (empty) instance from a class is so cheap that we can afford those extra CPU cycles to achieve cleaner code. I'm fine with either way. :-)

Comment (kpanwar) [inline src/MSAL.PCL/Internal/Http/HttpClientFactory.cs]: creating HttpClient instances could result in out of sockets exception. guidance says that create a single instance and use it for the lifetime of the application.

Comment (rayluo) [inline src/MSAL.PCL/Internal/Http/HttpClientFactory.cs]: You are right (again). [HttpClient has been designed to be re-used for multiple calls.](http://stackoverflow.com/questions/22560971/what-is-the-overhead-of-creating-a-new-httpclient-per-call-in-a-webapi-client)
Learn something new (to me) today.

File: src/MSAL.PCL/Internal/Http/HttpClientFactory.cs (hunk 3 above)
Comment (SomkaPe) [inline src/MSAL.PCL/Internal/Http/HttpClientFactory.cs]: done

**Pull request merged**